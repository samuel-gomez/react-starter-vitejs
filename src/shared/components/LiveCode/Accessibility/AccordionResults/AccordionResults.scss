@import '@axa-fr/react-toolkit-core/src/common/scss/core.scss';

.af-accordion {
  &--accessibility {
    min-height: 300px;
    &-wrapper-body {
      margin-left: 2rem;
      a {
        text-decoration: underline;
      }
      .af-badge {
        margin-right: 0.5rem;
      }
    }
  }
}
.af-accordion--light .af-accordion__item {
  border-bottom: solid 1px $color-gray-1;
}
