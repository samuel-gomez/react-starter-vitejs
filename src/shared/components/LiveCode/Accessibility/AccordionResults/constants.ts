const colorTags = {
  critical: 'error',
  serious: 'danger',
  minor: 'info',
  moderate: 'info',
};

export type TcolorTags = (typeof colorTags)[keyof typeof colorTags];

export default colorTags;
