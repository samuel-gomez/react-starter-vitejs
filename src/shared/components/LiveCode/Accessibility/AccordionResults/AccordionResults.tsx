import { Accordion, Badge, CollapseCard } from '@axa-fr/react-toolkit-all';
import { type Result } from 'axe-core';
import './AccordionResults.scss';
import colorTags from './constants';

const AccordionResults = ({ results }: { results: Result[] }) => (
  <Accordion classModifier="light accessibility" onlyOne={false}>
    {results.map((result: Result) => (
      <CollapseCard key={result.id} isOpen={false} id={result.id}>
        <CollapseCard.Header key={result.id}>
          {result.description} {result.impact && <Badge classModifier={colorTags[result.impact]}>{result.impact}</Badge>}
        </CollapseCard.Header>
        <CollapseCard.Body key={result.id}>
          <div className="af-accordion--accessibility-wrapper-body">
            <p>{result.help}</p>
            <p>
              <a href={result.helpUrl} rel="noreferrer" target="_blank">
                Plus d&apos;infos...
              </a>
            </p>
          </div>
          <div className="af-accordion--accessibility-wrapper-body">
            <span>Tags : </span>
            {result.tags.map((tag: string) => (
              <Badge key={tag} classModifier="info">
                {tag}
              </Badge>
            ))}
          </div>
        </CollapseCard.Body>
      </CollapseCard>
    ))}
  </Accordion>
);

export default AccordionResults;
