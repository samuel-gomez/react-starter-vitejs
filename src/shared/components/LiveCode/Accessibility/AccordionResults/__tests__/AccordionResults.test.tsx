import { render } from '@testing-library/react';
import { type Result } from 'axe-core';
import AccordionResults from '../AccordionResults';

const resultsFnMock = [
  {
    description: 'test',
    help: 'test',
    helpUrl: 'url',
    impact: 'critical',
    tags: ['tag1'],
    id: 'id-test',
  },
];

describe('<AccordionResults />', () => {
  it('Render <AccordionResults /> with results', () => {
    const { asFragment } = render(<AccordionResults results={resultsFnMock as Result[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
