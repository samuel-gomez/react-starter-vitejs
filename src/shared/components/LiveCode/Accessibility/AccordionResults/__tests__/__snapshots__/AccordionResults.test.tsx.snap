// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AccordionResults /> > Render <AccordionResults /> with results 1`] = `
<DocumentFragment>
  <div
    aria-multiselectable="true"
    class="af-accordion af-accordion--light af-accordion--accessibility"
    role="tablist"
  >
    <div
      class="af-accordion__item"
    >
      <div
        class="af-accordion__item-header"
        id="id-test"
        role="tab"
      >
        <a
          aria-controls="id-testbody"
          aria-expanded="false"
          class="af-accordion__item-toggle"
          data-toggle="isOpen"
          href="#id-testbody"
        >
          <h3
            class="af-accordion__item-title"
          >
            <span
              class="glyphicon glyphicon-menu-down"
            />
            test 
            <span
              class="af-badge af-badge--error"
            >
              critical
            </span>
          </h3>
        </a>
      </div>
      <div
        aria-labelledby="id-test"
        class="af-accordion__collapse af-accordion__collapse--close"
        id="id-testbody"
        role="tabpanel"
        style="max-height: 0;"
      >
        <div
          class="af-accordion__block"
        >
          <div
            class="af-accordion--accessibility-wrapper-body"
          >
            <p>
              test
            </p>
            <p>
              <a
                href="url"
                rel="noreferrer"
                target="_blank"
              >
                Plus d'infos...
              </a>
            </p>
          </div>
          <div
            class="af-accordion--accessibility-wrapper-body"
          >
            <span>
              Tags : 
            </span>
            <span
              class="af-badge af-badge--info"
            >
              tag1
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
